lib/screens/contributors.dart

```diff
@@ -295,16 +356,50 @@ Future<List<ContributorCard>> getContributors(
   return contriCards;
 }
 
-addToContributors() {
-  // TODO: Trigger an alert box or something similar
-  // Ask the user to enter the details
-  // Adds the user to contributors list
+void addToContributors(BuildContext context,
+    Function(ContributorCard) addContributorCallback) async {
+  // Fetch contributor data from GitHub based on the provided username
+  String githubUsername =
+      'githubUsername'; // Replace with the actual GitHub username
+
+  // Fetch contributor data from GitHub using the GitHub API
+  final githubResponse =
+      await http.get(Uri.https('api.github.com', '/users/$githubUsername'));
+  if (githubResponse.statusCode == 200) {
+    // Parse the GitHub response to extract contributor details
+    final contributorData = contributorDetailFromJson(githubResponse.body);
+
+    ContributorCard newContributor = ContributorCard(
+      userName: contributorData.login,
+      name: contributorData.name ?? '',
+      desc: contributorData.bio ?? '',
+      location: contributorData.location ?? '',
+      twitterUsername: contributorData.twitterUsername ?? '',
+      displayImgUrl: contributorData.avatarUrl,
+      website: contributorData.blog.isEmpty
+          ? contributorData.htmlUrl
+          : contributorData.blog,
+    );
+
+    // Access cardList and update it within the _ContributorsState instance
+    addContributorCallback(newContributor);
+
+    try {
+      Navigator.of(context).pop(); // Close the dialog
+    } catch (e) {
+      // Handle the error gracefully (e.g., log it)
+      print('Error when popping Navigator: $e');
+    }
+  } else {
+    // Handle the case where GitHub data could not be retrieved
+    print('Failed to fetch GitHub data for the contributor.');
+  }
 }
 
 _launchURL(String gurl) async {
   String url = gurl;
-  if (await canLaunchUrl(Uri.parse(url))) {
-    await launchUrl(Uri.parse(url));
+  if (await canLaunch(url)) {

```

Comment:
![image](https://github.com/IEEE-VIT/hacktoberfest-flutter/assets/84137299/07846619-6e6f-4c21-86e2-7583a2569782)

Please stick to newer methods and functions